src/rez/plugin_managers.py

```diff
@@ -52,18 +53,23 @@ def extend_path(path, name):
     init_py = "__init__" + os.extsep + "py"
     path = path[:]
 
-    for dir_ in config.plugin_path:
-        if not os.path.isdir(dir_):
-            if config.debug("plugins"):
-                print_debug("skipped nonexistant rez plugin path: %s" % dir_)
-            continue
+    def append_if_valid(dir_):
+        if os.path.isdir(dir_):
+            subdir = os.path.normcase(os.path.join(dir_, pname))
+            initfile = os.path.join(subdir, init_py)
+            if subdir not in path and os.path.isfile(initfile):
+                path.append(subdir)
 
-        subdir = os.path.join(dir_, pname)
-        # XXX This may still add duplicate entries to path on
-        # case-insensitive filesystems
-        initfile = os.path.join(subdir, init_py)
-        if subdir not in path and os.path.isfile(initfile):
-            path.append(subdir)
+        elif config.debug("plugins"):
+            print_debug("skipped nonexistant rez plugin path: %s" % dir_)
+
+    # Extend old-style plugins
+    for dir_ in config.plugin_path:
+        append_if_valid(dir_)
+    # Extend new-style plugins
+    for loader, name, ispkg in pkgutil.iter_modules():
```

Comment:
Deal with plugins with the same name here. If there's a clash, new-style plugin should take precedence, and a warning message should be printed. This will help migration.

Reply:
Ps - this will also match existing behaviour wrt a recent change, which ensured that plugins with the same name as a builtin (such as package_repository.filesystem) override the builtin.

Reply:
> Deal with plugins with the same name here.

Hmmm, I think we should deal it in `RezPluginType.load_plugins`, since here we only get the python module name, not the name of each plugin.

Reply:
warning message added in 8e820b6